src/snowflake/snowpark/relational_grouped_dataframe.py

```diff
@@ -365,6 +366,103 @@ def end_partition(self, pdf: pandas.DataFrame) -> pandas.DataFrame:
 
     applyInPandas = apply_in_pandas
 
+    def pivot(
+        self, pivot_col: ColumnOrName, values: Optional[Iterable[LiteralType]] = None
+    ) -> "RelationalGroupedDataFrame":
+        """Rotates this DataFrame by turning unique values from one column in the input
+        expression into multiple columns and aggregating results where required on any
+        remaining column values.
+
+        When ``values`` argument is not specified, distinct values from pivot column are
+        inferred by calculating the distinct values in the pivot column. This is a less
+        efficient method and therefore not recommended.
+
+        Only one aggregate is supported with pivot.
+
+        Args:
+            pivot_col: The column or name of the column to use.
+            values: An optional list of values in the column.
+
+        Example::
+
+            >>> create_result = session.sql('''create or replace temp table monthly_sales(empid int, team text, amount int, month text)
+            ... as select * from values
+            ... (1, 'A', 10000, 'JAN'),
+            ... (1, 'B', 400, 'JAN'),
+            ... (2, 'A', 4500, 'JAN'),
+            ... (2, 'A', 35000, 'JAN'),
+            ... (1, 'B', 5000, 'FEB'),
+            ... (1, 'A', 3000, 'FEB'),
+            ... (2, 'B', 200, 'FEB') ''').collect()
+            >>> df = session.table("monthly_sales")
+            >>> df.group_by("empid").pivot("month", ['JAN', 'FEB']).sum("amount").sort(df["empid"]).show()
+            -------------------------------
+            |"EMPID"  |"'JAN'"  |"'FEB'"  |
+            -------------------------------
+            |1        |10400    |8000     |
+            |2        |39500    |200      |
+            -------------------------------
+            <BLANKLINE>
+
+        Example::
+
+            >>> create_result = session.sql('''create or replace temp table monthly_sales(empid int, team text, amount int, month text)
+            ... as select * from values
+            ... (1, 'A', 10000, 'JAN'),
+            ... (1, 'B', 400, 'JAN'),
+            ... (2, 'A', 4500, 'JAN'),
+            ... (2, 'A', 35000, 'JAN'),
+            ... (1, 'B', 5000, 'FEB'),
+            ... (1, 'A', 3000, 'FEB'),
+            ... (2, 'B', 200, 'FEB') ''').collect()
```

Comment:
I wonder if we could use `create_dataframe` then `save_as_table` to represent this `session.sql`? Maybe it is not worth it but just thinking out loud.